pkg/storage/iscsiNode.go

```diff
@@ -70,15 +78,37 @@ func (iscsi *iscsiStorage) NodePublishVolume(ctx context.Context, req *csi.NodeP
 	}
 
 	volumeName, _ := common.VolumeIdGetName(req.GetVolumeId())
-	klog.Infof("publishing volume %s", volumeName)
-	klog.Infof("target path: %s", req.GetTargetPath())
+	wwn, _ := common.VolumeIdGetWwn(req.GetVolumeId())
+
+	// Ensure that NodePublishVolume and NodeUnpublishVolume are only called once per volume
+	if mu, present := gateKeepers[volumeName]; present {
```

Comment:
Technically, Go maps are not thread-safe, so there's a race condition here that lead the Go runtime to panic.  I think you need a global mutex to protect access to the map.